src/_pytest/fixtures.py

```diff
@@ -1052,6 +973,9 @@ def __init__(
         self.cached_result: Optional[_FixtureCachedResult[FixtureValue]] = None
         self._finalizers: Final[List[Callable[[], object]]] = []
 
+        # Whether fixture is a pseudo-fixture made in direct parametrizations.
```

Comment:
I understand this terminology is preexisting, but I don't like it much, "pseudo" is not very descriptive... Also, there's actually a `PseudoFixtureDef` already which is something else entirely and which only adds to the confusion :)

Anyway we can clean the terms later, you don't need to change it.

Reply:
While we are at it, perhaps `is_parametrized` would be a good fit?

Reply:
Also do we want this to be public, or prefix it with `_` to be sure?

Reply:
How is to introduce an `IdentityFixture` class and use `isinstance` instead of adding and checking `is_pseudo`?

```python
class IdentityFixture(FixtureDef):
    def __init__(fixturemanager, argname, scope):
        super().__init__(fixturemanager, "", argname, lambda request: request.param, scope, None, _ispytest=True)
```
This also accounts for @RonnyPfannschmidt 's [comment](https://github.com/pytest-dev/pytest/pull/11220/files#r1268521442). Just we should remove `@final` from `FixtureDef` class.

Or simply introduce `is_identity`? @nicoddemus , Couldn't `is_parametrized` be confusing, since such a fixture is not really parametrized (using `@pytest.fixture(params=)` or `@pytest.mark.parametrize(indirect=True)` but the test is directly parametrized?

Reply:
Hmm that's a good suggestion; my gut reaction is that I like this idea of having `IdentityFixture` (or some other name in case others have a suggestion). Let's see what others think.

Reply:
i'd like to note that we do have a "spagetty-fied" pattern here about configuring a "dependency injection container" with either direct values or parameters to "factories"

if we introduce polymorphism, we might want to make "identity" and "from factory, possibly with parameters" siblings and check if we can get rid of those is_pseudo checks altogether

there is a rat-tail of discussions to be had around this, so we ought to consider it but without going down the rabbit  hole too deep

Reply:
As `params` arg of this very fixturedef is `None`, it's automatically ruled out in the only place (`fixtures.py::pytest_generate_tests`) where I used `is_pseudo`. So the need for this attribute is eliminated.